File: OOP/Homeworks/OOPInJava/Geometry/classes/shapes/planeShapes/Triangle.java
package classes.shapes.planeShapes;

import java.util.ArrayList;

import classes.Vertex;
import classes.shapes.PlaneShape;

public class Triangle extends PlaneShape {
	ArrayList<Vertex> vertices = new ArrayList<Vertex>();
	
	public Triangle(ArrayList<Vertex> vertices) {
		super(vertices);
		this.setTriangle(vertices);
	}

    public void setTriangle(ArrayList<Vertex> vertices) {
        this.vertices = vertices;
    }
    
	public double getArea() {
		double sideOne = getTringleSides()[0];
		double sideTwo = getTringleSides()[1];
		double sideThree = getTringleSides()[2];
		
		double halfPerimeter = getPerimeter() / 2;
		
		double area = Math.sqrt(halfPerimeter * (halfPerimeter - sideOne) * (halfPerimeter - sideTwo) * (halfPerimeter - sideThree));
		
		return area;
	}
	
	public double getPerimeter() {
		double sideOne = getTringleSides()[0];
		double sideTwo = getTringleSides()[1];
		double sideThree = getTringleSides()[2];
		
		double perimeter = sideOne + sideTwo + sideThree;
		
		return perimeter;
	}
	
	public double[] getTringleSides() {
		Vertex vertexOne = vertices.get(0);
		Vertex vertexTwo = vertices.get(1);
		Vertex vertexThree = vertices.get(2);
		double sideOne = Math.sqrt(Math.pow((vertexOne.getX() - vertexTwo.getX()), 2) + Math.pow((vertexOne.getY() - vertexTwo.getY()), 2));
		double sideTwo = Math.sqrt(Math.pow((vertexOne.getX() - vertexThree.getX()), 2) + Math.pow((vertexOne.getY() - vertexThree.getY()), 2));
		double sideThree = Math.sqrt(Math.pow((vertexTwo.getX() - vertexThree.getX()), 2) + Math.pow((vertexTwo.getY() - vertexThree.getY()), 2));
		
		double[] triangleSides = new double[3];
		triangleSides[0] = sideOne;
		triangleSides[1] = sideTwo;
		triangleSides[2] = sideThree;
		
		return triangleSides;
	}
}

File: C#Basics/Exams/Exam12April2014Morning/Triangle/Triangle.cs
namespace Triangle
{
    using System;

    class Triangle
    {
        static void Main()
        {
            int aX = int.Parse(Console.ReadLine());
            int aY = int.Parse(Console.ReadLine());
            int bX = int.Parse(Console.ReadLine());
            int bY = int.Parse(Console.ReadLine());
            int cX = int.Parse(Console.ReadLine());
            int cY = int.Parse(Console.ReadLine());
            double distanceAB = Math.Sqrt(Math.Pow(bX - aX, 2) + Math.Pow(bY - aY, 2));
            double distanceBC = Math.Sqrt(Math.Pow(cX - bX, 2) + Math.Pow(cY - bY, 2));
            double distanceCA = Math.Sqrt(Math.Pow(aX - cX, 2) + Math.Pow(aY - cY, 2));

            if (distanceAB + distanceBC > distanceCA &&
                distanceBC + distanceCA > distanceAB &&
                distanceAB + distanceCA > distanceBC)
            {
                double halfPerimeter = (distanceAB + distanceBC + distanceCA) / 2;
                double area = Math.Sqrt(
                    halfPerimeter *
                    (halfPerimeter - distanceAB) *
                    (halfPerimeter - distanceBC) *
                    (halfPerimeter - distanceCA));

                Console.WriteLine("Yes\n{0:0.00}", area);
            }
            else
            {
                Console.WriteLine("No\n{0:0.00}", distanceAB);
            }
        }
    }
}
